components/infobox/commons/infobox_player.lua

```diff
@@ -0,0 +1,267 @@
+local Class = require('Module:Class')
+local Cell = require('Module:Infobox/Cell')
+local Infobox = require('Module:Infobox')
+local Template = require('Module:Template')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+local Namespace = require('Module:Namespace')
+local Links = require('Module:Links')
+local Flags = require('Module:Flags')
+--local GetBirthAndDeath = require('Module:???')._get
+
+--the following 3 lines as a temp workaround until the Birth&Death stuff is implemented:
+local function GetBirthAndDeath()
+    return '', nil, nil, nil
+end
+
+local getArgs = require('Module:Arguments').getArgs
+
+local Player = Class.new()
+local Language = mw.language.new('en')
+local _LINK_VARIANT = 'player'
+local shouldStoreData
+
+function Player.run(frame)
+    return Player:createInfobox(frame)
+end
+
+function Player:createInfobox(frame)
+    local args = getArgs(frame)
+    self.frame = frame
+    self.pagename = mw.title.getCurrentTitle().text
+    self.name = args.id or self.pagename
+
+    shouldStoreData = Player:shouldStoreData(args)
+
+    if args.game == nil then
+        return error('Please provide a game!')
+    end
+
+    local infobox = Infobox:create(frame, args.game)
+
+    local earnings = Player:calculateEarnings(args)
+    Variables.varDefine('earnings', earnings)
+    if earnings == 0 then
+        earnings = nil
+    else
+        earnings = '$' .. Language:formatNum(earnings)
+    end
+    local nameDisplay = Player:nameDisplay(args)
+    local role = Player:getRole(args)
+    local birthDisplay, deathDisplay, birthday, deathday
+        = GetBirthAndDeath(
+            args.birth_date,
+            args.birth_location,
+            args.death_date,
+            role.category,
+            shouldStoreData
+        )
+    local status = Player:getStatus(args)
+
+    infobox :name(nameDisplay)
+            :image(args.image, args.defaultImage)
+            :centeredCell(args.caption)
+            :header('Player Information', true)
+            :cell('Name', args.name)
+            :cell('Romanized Name', args.romanized_name)
+            :cell('Birth', birthDisplay)
+            :cell('Died', deathDisplay)
+            :cell('Status', status.display)
+            :cell(role.title or 'Role', role.display)
+            :fcell(Cell :new('Country')
+                        :options({})
+                        :content(
+                            Player:_createLocation(args.country or args.nationality, args.location),
+                            Player:_createLocation(args.country2 or args.nationality2, args.location2),
+                            Player:_createLocation(args.country3 or args.nationality3, args.location3)
```

Comment:
You can use `unpack` for this, combined with `content`'s varargs. See #186's implementation of organizers.